session.go

```diff
@@ -44,12 +44,11 @@ type Session struct {
 	availableTestHelpers map[string][]model.Service
 	byteCounter          *bytecounter.Counter
 	httpDefaultTransport httptransport.RoundTripper
-	httpNoProxyTransport httptransport.RoundTripper
 	kvStore              model.KeyValueStore
 	privacySettings      model.PrivacySettings
-	explicitProxy        bool
 	location             *model.LocationInfo
 	logger               model.Logger
+	proxyURL             *url.URL
```

Comment:
The transport w/o proxy is gone. The rationale of this change is the following:

1. honouring `HTTP_PROXY` is increasingly weird when we have HTTP and non-HTTP pieces of code

2. so, let's instead introduce a concept of explicit proxy at SOCKS5 level

3. if we have a concept of explicit proxy, then use it for everything, to honour the user wishes

4. then we don't need a transport without proxy anymore